sdk/storage/src/blob/clients/container_client.rs

```diff
@@ -107,3 +107,88 @@ impl ContainerClient {
             .prepare_request(url, method, http_header_adder, request_body)
     }
 }
+
+#[cfg(test)]
+#[cfg(feature = "test_integration")]
+mod integration_tests {
+    use super::*;
+    use crate::{blob::clients::AsBlobClient, core::prelude::*};
+    use url::Url;
+
+    fn get_emulator_client(container_name: &str) -> Arc<ContainerClient> {
+        let blob_storage_url =
+            Url::parse("http://127.0.0.1:10000").expect("the default local storage emulator URL");
+        let queue_storage_url =
+            Url::parse("http://127.0.0.1:10001").expect("the default local storage emulator URL");
+        let table_storage_url =
+            Url::parse("http://127.0.0.1:10002").expect("the default local storage emulator URL");
+        let filesystem_url =
+            Url::parse("http://127.0.0.1:10004").expect("the default local storage emulator URL");
+
+        let http_client: Arc<dyn HttpClient> = Arc::new(reqwest::Client::new());
+        let storage_account = StorageAccountClient::new_emulator(
+            http_client,
+            &blob_storage_url,
+            &table_storage_url,
+            &queue_storage_url,
+            &filesystem_url,
+        )
+        .as_storage_client();
+
+        storage_account.as_container_client(container_name)
+    }
+
+    #[tokio::test]
+    async fn test_create_delete() {
+        let container_name = uuid::Uuid::new_v4().to_string();
+        let container_client = get_emulator_client(&container_name);
+
+        container_client
+            .create()
+            .execute()
+            .await
+            .expect("create container should succeed");
+        container_client
+            .delete()
+            .execute()
+            .await
+            .expect("delete container should succeed");
+    }
+
+    #[tokio::test]
+    async fn test_list_blobs() {
```

Comment:
This new test `test_list_blobs` is being flaky for me locally and with integration test builds. It fails sometimes with an internal error in Azure Storage Emulator.

```
---- blob::clients::container_client::integration_tests::test_list_blobs stdout ----
thread 'blob::clients::container_client::integration_tests::test_list_blobs' panicked at 'put block blob should succeed: UnexpectedStatusCode { expected: [201], received: 500, body: "\u{feff}<?xml version=\"1.0\" encoding=\"utf-8\"?><Error><Code>InternalError</Code><Message>Server encountered an internal error. Please try again after some time.\nRequestId:12be0ec4-2e4b-4ccd-a648-8b21856cadf2\nTime:2021-06-11T22:05:43.8264505Z</Message></Error>" }', sdk\storage\src\blob\clients\container_client.rs:175:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    blob::clients::container_client::integration_tests::test_list_blobs
```

Reply:
It is not just that test. I've had other fail just sometimes with the same internal error:
test table::clients::entity_client::integration_tests::test_insert_or_merge ... FAILED
test blob::clients::blob_client::integration_tests::test_get_properties ... FAILED

Reply:
Obviously there is a problem, but I cannot see how it could be on the client side. I am starting to doubt the windows emulator is reliable. Again, I am working with Azurite on Linux.

Reply:
I think Azure Storage Emulator should be replaced with Azurite. I added #295 & #296 to fix issues for running on Azurite. Once those two are merged, we can disabled the transaction test #297 in order to port over to Azurite. Once testing is moved over, tests here should pass.

Reply:
Please merge in master branch after #299 is merged to use Azurite.

Reply:
Will do.